InterfacePython/TasmanianParticleSwarm.py

```diff
@@ -0,0 +1,264 @@
+type_optim_dom_fn = CFUNCTYPE(c_bool, c_int, c_double_p)
+type_dream_random = CFUNCTYPE(c_double)
+
+pLibDTSG = cdll.LoadLibrary(__path_libdream__)
+
+pLibDTSG.tsgParticleSwarmState_Construct.restype = c_void_p
+pLibDTSG.tsgParticleSwarmState_Construct.argtypes = [c_int, c_int]
+
+pLibDTSG.tsgParticleSwarmState_Destruct.argtypes = [c_void_p]
+
+pLibDTSG.tsgParticleSwarmState_GetNumDimensions.restype = c_int
+pLibDTSG.tsgParticleSwarmState_GetNumDimensions.argtype = [c_void_p]
+pLibDTSG.tsgParticleSwarmState_GetNumParticles.restype = c_int
+pLibDTSG.tsgParticleSwarmState_GetNumParticles.argtype = [c_void_p]
+pLibDTSG.tsgParticleSwarmState_GetParticlePositions.argtypes = [c_void_p, np_double_arr]
+pLibDTSG.tsgParticleSwarmState_GetParticleVelocities.argtypes = [c_void_p, np_double_arr]
+pLibDTSG.tsgParticleSwarmState_GetBestParticlePositions.argtypes = [c_void_p, np_double_arr]
+pLibDTSG.tsgParticleSwarmState_GetBestPosition.argtypes = [c_void_p, np_double_arr]
+
+pLibDTSG.tsgParticleSwarmState_SetParticlePositions.argtypes = [c_void_p, np_double_arr]
+pLibDTSG.tsgParticleSwarmState_SetParticleVelocities.argtypes = [c_void_p, np_double_arr]
+pLibDTSG.tsgParticleSwarmState_SetBestParticlePositions.argtypes = [c_void_p, np_double_arr]
+
+pLibDTSG.tsgParticleSwarmState_ClearBestParticles.argtypes = [c_void_p]
+pLibDTSG.tsgParticleSwarmState_ClearCache.argtypes = [c_void_p]
+pLibDTSG.tsgParticleSwarmState_InitializeParticlesInsideBox.argtypes = [c_void_p, np_double_arr, np_double_arr, c_char_p, c_int, type_dream_random]
+pLibDTSG.tsgParticleSwarm.argtypes = [type_optim_obj_fn, c_int, type_optim_dom_fn, c_void_p, c_double, c_double, c_double, c_char_p, c_int, type_dream_random]
+
+class ParticleSwarmState:
+    '''
+    Wrapper to class TasOptimization::ParticleSwarmState
+    '''
+    def __init__(self, iNumDimensions, iNumParticles):
+        '''
+        Constructs a new state with a given number of dimensions and particles.
+
+        iNumDimensions : positive integer indicating the number of dimensions.
+        iNumParticles  : positive integer indicating the number of particles.
+        '''
+        self.TasmanianParticleSwarmState = True
+        self.pStatePntr = c_void_p(pLibDTSG.tsgParticleSwarmState_Construct(iNumDimensions, iNumParticles))
+
+    def __del__(self):
+        '''
+        Deletes an instance of the particle swarm state.
+        '''
+        pLibDTSG.tsgParticleSwarmState_Destruct(self.pStatePntr)
+
+    def getNumDimensions(self):
+        '''
+        Return the number of dimensions.
+        '''
+        return pLibDTSG.tsgParticleSwarmState_GetNumDimensions(self.pStatePntr)
+
+    def getNumParticles(self):
+        '''
+        Return the number of particles.
+        '''
+        return pLibDTSG.tsgParticleSwarmState_GetNumParticles(self.pStatePntr)
+
+    def getParticlePositions(self):
+        '''
+        Return the particle positions as a NumPy array.
```

Comment:
Describe the array, i.e., it is a 2D array with rows corresponding to one thing and columns corresponding to another.

Reply:
Changed here and in similar functions.